game/lib/model/event_model.dart

```diff
@@ -11,12 +11,24 @@ class EventModel extends ChangeNotifier {
 
   EventModel(ApiClient client) : _client = client {
     client.clientApi.updateEventDataStream.listen((event) {
-      if (!event.event is String) {
-        searchResults = event.event;
+      if (!(event.event is String)) {
+        if (!_events.containsKey(event.event.id)) {
+          if (searchResults != null) {
```

Comment:
This is not easy to follow as well but this could be by design of our API. I suggest an improvement (easy one) for the API to clean up this logic:

Currently, this is what 
UpdateEventDataDto {
  event: EventDto | string;
  deleted: boolean;
} 
looks like.

UpdateEventDataDto {
  event: EventDto | string;
  isSearchResult: boolean; // This will be true when requestAllEventData is called
  deleted: boolean;
} .

Now instead of searchResults being nullable, simply make it an empty list and clear it every time the frontend search function (the one up above) is called. Keep the deletion/update logic the same. But, whenever isSearchResult is true, simply add that event to the searchResults list. This will clean up the logic a lot.

Reply:
I cleaned up my code in a different way without changing the dto. Hopefully it's a bit clearer now